_episodes/02-keras.md

```diff
@@ -0,0 +1,514 @@
+> > The other two seem to be most easily separated by a combination of bill length and bill
+> > depth.
+> {:.solution}
+{:.challenge}
+
+### Input and Output Selection
+Now that we have familiarized ourselves with the dataset we can select the data attributes to use
+as input for the neural network and the target that we want to predict.
+
+In the rest of this episode we will use the `bill_length_mm`, `bill_depth_mm`, `flipper_length_mm`, `body_mass_g` attributes.
+The target for the classification task will be the `species`.
+
+Using the following code we can select these columns from the dataframe:
+~~~
+training_data = penguins.drop(columns=["species", 'island', 'sex'])
+target = penguins['species']
+~~~
+{:.language-python}
+
+> ## Data Exploration
+> Exploring the data is an important step to familiarize yourself with the problem and to help you
+> determine the relavent inputs and outputs.
+{:.keypoints}
+## 3. Prepare data
+The input data and target data are not yet in a format that is suitable to use for training a neural network.
+During the exploration phase you may have noticed that some rows in the dataset have missing (NaN)
+values, leaving such values in the input data will ruin the training, so we need to deal with them.
+There are many ways to deal with missing values, but for now we will just remove the offending rows.
+
+Add a call to `dropna()` before the input_data definition that we used above, the cell should now
+look like this:
+~~~
+valid_data = penguins.dropna()
+
+training_data = valid_data.drop(columns=["species", 'island', 'sex'])
+target = valid_data['species']
+~~~
+{:.language-python}
+
+Second, the target data is also in a format that cannot be used to train.
+To train a neural network we need to be able to calculate how "far away" the species predicted by the
+neural network is from the true species.
+When the target is a string category column as we have here it is very difficult to determine this "distance" or error.
+Therefore we will transform this column into a more suitable format.
+Again there are many ways to do this, however we will be using the 1-hot encoding.
+This encoding creates multiple columns, as many as there are unique values, and
+puts a 1 in the column with the corresponding correct class, and 0's in
+the other columns.
+For instance, for a penguin of the Adelie species the 1 hot encoding would be 1 0 0
+
+Fortunately pandas is able to generate this encoding for us.
+~~~
+target = pd.get_dummies(valid_data['species'])
+target.head() # print out the top 5 to see what it looks like.
+~~~
+{:.language-python}
+
+> ## Keras for Neural Networks
+> For this lesson we will be using [Keras](https://keras.io/) to define and train our neural network
+> models.
+> Keras is a machine learning framework with ease of use as one of its main features.
+> It is part of the tensorflow python package and can be imported using `from tensorflow import keras`.
```

Comment:
The way to import keras is mentioned here in the call-out, so it can easily be missed. Maybe repeat it before the first code block that uses Keras